### PR TITLE
MM-55440: initial version of feature usage

### PR DESCRIPTION
**Pull request by ifoukarakis**
#### Summary

Initial version of explore for feature usage.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-55440

### CODE REVIEW

File: models/product.model.lkml [hunk 1]
```diff
@@ -171,3 +171,29 @@ explore: fct_notification_stats {
...
+
+explore: fct_feature_daily_snapshot {
+  label: "Paid feature usage"
+  group_label: "[New] Feature usage"
+
+
+  join: dim_excludable_servers {
+    relationship: many_to_one
+    type: left_outer
+    sql_on: ${fct_feature_daily_snapshot.server_id}.server_id} = ${dim_excludable_servers.server_id} ;;
```
Comment (sri-byte) [inline models/product.model.lkml:184]: extra `.server_id}` here

File: models/product.model.lkml [hunk 2]
```diff
@@ -171,3 +171,29 @@ explore: fct_notification_stats {
...
+  }
+
+
+  join: dim_cloud_customers {
+    relationship: many_to_one
+    type: left_outer # We might not have customer telemetry for the NPS Score server
```
Comment (sri-byte) [inline models/product.model.lkml:190]: comment can be removed

File: models/product.model.lkml [hunk 3]
```diff
@@ -171,3 +171,29 @@ explore: fct_notification_stats {
...
+    sql_on: ${fct_feature_daily_snapshot.server_id} = ${dim_cloud_customers.server_id} ;;
+  }
+
+  join: dim_self_hosted_customers {
+    relationship: many_to_one
+    type: left_outer # We might not have customer telemetry for the NPS Score server
```
Comment (sri-byte) [inline models/product.model.lkml:196]: comment can be removed here too

File: views/marts/product/fct_feature_daily_snapshot.view.lkml [hunk 4]
```diff
@@ -0,0 +1,228 @@
+view: fct_feature_daily_snapshot {
+  sql_table_name: "MART_PRODUCT"."FCT_FEATURE_DAILY_SNAPSHOT" ;;
+
+
+  ## Keys
+
+  dimension: daily_server_id {
+    type: string
+    sql: ${TABLE}."DAILY_SERVER_ID" ;;
```
Comment (sri-byte) [inline views/marts/product/fct_feature_daily_snapshot.view.lkml:9]: nit: the column names have to be lower case, to be consistent with other views in the folder. 

`${TABLE}."DAILY_SERVER_ID"` to `${TABLE}.daily_server_id`

**Review by sri-byte — CHANGES_REQUESTED** [review]

File: views/marts/product/fct_feature_daily_snapshot.view.lkml [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+  measure: total_count_custom_ug_events_monthly {
+    type: sum
+    sql: ${count_custom_ug_events_monthly} ;;
+    label: "Total # of Custom User Group Events (Monthly) "
+    group_label: "* Metrics: Paid features"
+  }
+
+  measure: total_count_custom_ug_users_daily {
+    type: sum
+    sql: ${count_custom_ug_events_monthly} ;;
```
Comment (catalintomai) [inline views/marts/product/fct_feature_daily_snapshot.view.lkml:85]: count_custom_ug_events_monthly -> count_custom_ug_events_daily

File: views/marts/product/fct_feature_daily_snapshot.view.lkml [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+    label: "# of Guest Account Events (Monthly)"
+    description: "The number of events related to Guest Accounts feature  for the current day and past 29 days."
+    group_label: "* Features: Telemetry Information"
+  }
+
+  dimension: count_guest_accounts_users_daily {
+    type: number
+    sql: ${TABLE}.count_guest_accounts_users_daily ;;
+    label: "# of Guest Accounts Unique Users (Daily)"
+    description: "The number of unique uusers related to Guest Account feature for the current day."
```
Comment (catalintomai) [inline views/marts/product/fct_feature_daily_snapshot.view.lkml:119]: uusers -> users

File: views/marts/product/fct_feature_daily_snapshot.view.lkml [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+    group_label: "* Features: Telemetry Information"
+  }
+
+  dimension: count_guest_accounts_users_monthly {
+    type: number
+    sql: ${TABLE}.count_guest_accounts_users_daily ;;
```
Comment (catalintomai) [inline views/marts/product/fct_feature_daily_snapshot.view.lkml:125]: count_guest_accounts_users_daily -> count_guest_accounts_users_monthly

File: models/product.model.lkml (hunk 1 above)
Comment (ifoukarakis) [reply models/product.model.lkml]: Fixed.

File: views/marts/product/fct_feature_daily_snapshot.view.lkml (hunk 6 above)
Comment (ifoukarakis) [reply views/marts/product/fct_feature_daily_snapshot.view.lkml]: Fixed.

File: views/marts/product/fct_feature_daily_snapshot.view.lkml (hunk 5 above)
Comment (ifoukarakis) [reply views/marts/product/fct_feature_daily_snapshot.view.lkml]: Fixed.

File: views/marts/product/fct_feature_daily_snapshot.view.lkml (hunk 4 above)
Comment (ifoukarakis) [reply views/marts/product/fct_feature_daily_snapshot.view.lkml]: I did it for now, but as discussed in our meeting, let's decide if there's value in following the old practice.

File: models/product.model.lkml (hunk 2 above)
Comment (ifoukarakis) [reply models/product.model.lkml]: Removed.

**Review by catalintomai — APPROVED** [review]

**Review by sri-byte — APPROVED** [review]

**Pull request merged**